packages/monaco/src/browser/monaco-semantic-highlighting-service.ts

```diff
@@ -32,9 +33,51 @@ export class MonacoSemanticHighlightingService extends SemanticHighlightingServi
     @inject(EditorManager)
     protected readonly editorManager: EditorManager;
 
+    @inject(MonacoEditorService)
+    protected readonly monacoEditorService: MonacoEditorService;
+
     protected readonly decorations = new Map<string, Set<string>>();
     protected readonly toDisposeOnEditorClose = new Map<string, Disposable>();
 
+    // laguage id -> (scope index -> model decoration)
+    protected readonly modelDecorations = new Map<string, Map<number, monaco.editor.IModelDecorationOptions>>();
+    protected readonly decorationTypeKeys = new IdGenerator('DecorationType');
```

Comment:
please use another prefix, like ` MonacoSemanticHighlightingDecorationType` in order to avoid collisions